### ISSUE TITLE
IsMaximizable setting isn't respected

### ISSUE THREAD

**Issue by eduardobragaxz** [microsoft/microsoft-ui-xaml#8666]
### Describe the bug

Setting the OverlappedPresenter's IsMaximazable property to false only makes the caption button to be grayed out. You can still click it.

Related to an issue with maximized windows showing the resizable pointer on the upper edge, if you set IsResizable to false, the pointer will still change.

### Steps to reproduce the bug

1. Create a WinUI 3 project;
2. Get a reference to the window's presenter and set these properties to false;
3. Build and run the app.

### Expected behavior

It should be impossible to maximize the window.

### NuGet package version

Windows App SDK 1.4 Preview 1: 1.4.230628000-preview1

### Packaging type

Packaged (MSIX)

### Windows version

Windows 11 version 22H2 (22621, 2022 Update)

### IDE

Visual Studio 2022-preview

**Comment by Qiu233** [comment]
This issue is annoying, as it makes a window with `ExtendsContentIntoTitleBar=true` and `IsMaximizable=false` looks even worse when user accidentally double click on the titlebar.
Any fix for now?

**Comment by d1msk1y** [comment]
I've been tinkering for a long time with it and finally found a working and clean solution for it.

Basically you can make it work by intercepting the native window message for double-click (`WM_NCLBUTTONDBLCLK`) and ignore it.

### Steps
1. **Add these fields and imports to your MainWindow class:**
    ```csharp
    using WinRT.Interop;
    using System.Runtime.InteropServices;

    private const int WM_NCLBUTTONDBLCLK = 0x00A3;
    private IntPtr _hwnd;
    private SubclassDelegate? _newWndProc;
    private IntPtr _oldWndProc;
    private delegate IntPtr SubclassDelegate(IntPtr hWnd, uint msg, IntPtr wParam, IntPtr lParam);
    [DllImport("user32.dll")]
    private static extern IntPtr SetWindowLongPtr(IntPtr hWnd, int nIndex, SubclassDelegate newProc);
    [DllImport("user32.dll")]
    private static extern IntPtr CallWindowProc(IntPtr lpPrevWndFunc, IntPtr hWnd, uint msg, IntPtr wParam, IntPtr lParam);
    private const int GWLP_WNDPROC = -4;
    ```
2. **In your MainWindow constructor, after `InitializeComponent()`:**
    ```csharp
    _hwnd = WindowNative.GetWindowHandle(this);
    _newWndProc = WndProc;
    _oldWndProc = SetWindowLongPtr(_hwnd, GWLP_WNDPROC, _newWndProc);
    ```
3. **Add this method to your MainWindow class:**
    ```csharp
    private IntPtr WndProc(IntPtr hWnd, uint msg, IntPtr wParam, IntPtr lParam)
    {
        if (msg == WM_NCLBUTTONDBLCLK)
            return IntPtr.Zero; // Prevent maximize
        return CallWindowProc(_oldWndProc, hWnd, msg, wParam, lParam);
    }
    ```

tl;dr: Just copy the code above into your MainWindow class.

**Comment by thecco** [comment]
This still hasn’t been fixed. I’m still seeing the exact same issue.
Is WinUI 3 dead ?

**Comment by HO-COOH** [comment]
This seems to be fixed after winui 3 ~~1.8~~ 2.0.
